azure/Guidelines.md

```diff
@@ -477,6 +477,210 @@ Overall, this is a very brittle design that leads to a poor developer experience
+
+:white_check_mark: **DO** treat these query parameter names as case-sensitve.
+
+:no_entry: **DO NOT** prefix any of these query parameter names with "$" (the convention in the [OData standard](http://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part1-protocol.html#sec_QueryingCollections)).
+
+:no_entry: **DO NOT** define a _top_ query parameter to limit the number of results returned.  Use the _maxpagesize_ parameter for this purpose.
+
+:white_check_mark: **DO** apply the query options to the collection in the order shown in the table above.
+
+:white_check_mark: **DO** apply _select_ or _expand_ options after applying all the query options in the table above.
+
+#### filter
+
+:heavy_check_mark: **YOU MAY** support filtering of the results of a list operation with the _filter_ query parameter.
+
+The value of the _filter_ option is a Boolean expression which is evaluated for each resource in the collection, and only items where the expression evaluates to true are included in the response.
+
+:white_check_mark: **DO** omit all resources from the collection for which the _filter_ expression evaluates to false or to null, or references properties that are unavailable due to permissions.
+
+Example: return all Products whose Price is less than $10.00
+
+```http
+GET https://api.contoso.com/v1.0/products?filter=price lt 10.00
+```
+
+##### filter operators
+
+:ballot_box_with_check: **YOU SHOULD** support the following minimal set of operators in _filter_ expressions.
+
+Operator             | Description           | Example
+-------------------- | --------------------- | -----------------------------------------------------
+Comparison Operators |                       |
+eq                   | Equal                 | city eq 'Redmond'
+ne                   | Not equal             | city ne 'London'
+gt                   | Greater than          | price gt 20
+ge                   | Greater than or equal | price ge 10
+lt                   | Less than             | price lt 20
+le                   | Less than or equal    | price le 100
+Logical Operators    |                       |
+and                  | Logical and           | price le 200 and price gt 3.5
+or                   | Logical or            | price le 3.5 or price gt 200
+not                  | Logical negation      | not price le 3.5
+Grouping Operators   |                       |
+( )                  | Precedence grouping   | (priority eq 1 or city eq 'Redmond') and price gt 100
+
+:white_check_mark: **DO** respond with an error message as defined in the [Unsupported Requests](??) section if a client includes an operator in a _filter_ expression that is not supported by the operation.
+
+:white_check_mark: **DO** use the following operator precedence for supported operators when evaluating _filter_ expressions. Operators are listed by category in order of precedence from highest to lowest. Operators in the same category have equal precedence:
```

Comment:
Is there any easy way to slice the OData spec so we can say what we support without providing these sorts of details?  I worry that this is a slippery slope and would prefer to avoid BNF grammars for OData subsets in our guidelines.

Reply:
https://github.com/microsoft/api-guidelines/issues/247